internal/store/certificatesigningrequest.go

```diff
@@ -267,3 +268,35 @@ func (s *CertificateSigningRequestStore) UpdateConditions(ctx context.Context, o
 		return s.updateConditions(orgId, name, conditions)
 	})
 }
+
+func (s *CertificateSigningRequestStore) updateCertificate(orgId uuid.UUID, name string, certificate []byte) (bool, error) {
+	existingRecord := model.CertificateSigningRequest{Resource: model.Resource{OrgID: orgId, Name: name}}
+	result := s.db.First(&existingRecord)
+	if result.Error != nil {
+		return false, flterrors.ErrorFromGormError(result.Error)
+	}
+
+	if existingRecord.Status == nil {
+		existingRecord.Status = model.MakeJSONField(api.CertificateSigningRequestStatus{})
+	}
+	existingRecord.Status.Data.Certificate = &certificate
+
+	result = s.db.Model(existingRecord).Where("resource_version = ?", lo.FromPtr(existingRecord.ResourceVersion)).Updates(map[string]interface{}{
```

Comment:
I wonder if we can create update() function that will get object and push it to db while validating resource version cause it seems like same code is duplicate over different functions like here 
https://github.com/flightctl/flightctl/pull/491/files#diff-ee6d72cabf35aed3a2691f1505a707f49a38e752478c98a66d1dfc5ec6b16e51R252

Reply:
Are you suggesting to update the Resource Version only once between Approval and setting the Certificate? That would be an easy change

Reply:
Oh I think you are talking about abstracting over the various updates. I agree that would be a good change, likely out of scope for this PR though

Reply:
Nope, i suggest to have one function that will run 
`s.db.Model(existingRecord).Where("resource_version = ?", lo.FromPtr(existingRecord.ResourceVersion)).Updates(map[string]interface{}{...`
as seems like there is code duplication for the same logic between different functions